packages/mdc-textfield/foundation.ts

```diff
@@ -345,115 +315,101 @@ class MDCTextFieldFoundation extends MDCFoundation {
-  isDisabled() {
+  isDisabled(): boolean {
     return this.getNativeInput_().disabled;
   }
 
   /**
-   * @param {boolean} disabled Sets the text-field disabled or enabled.
+   * @param disabled Sets the text-field disabled or enabled.
    */
-  setDisabled(disabled) {
+  setDisabled(disabled: boolean): void {
     this.getNativeInput_().disabled = disabled;
     this.styleDisabled_(disabled);
   }
 
   /**
-   * @param {string} content Sets the content of the helper text.
+   * @param content Sets the content of the helper text.
    */
-  setHelperTextContent(content) {
+  setHelperTextContent(content: string): void {
     if (this.helperText_) {
       this.helperText_.setContent(content);
     }
   }
 
-  /**
-   * Sets character counter values that shows characters used and the total character limit.
-   * @param {number} currentLength
-   * @private
-   */
-  setCharacterCounter_(currentLength) {
-    if (!this.characterCounter_) return;
-
-    const maxLength = this.getNativeInput_().maxLength;
-    if (maxLength === -1) {
-      throw new Error('MDCTextFieldFoundation: Expected maxlength html property on text input or textarea.');
-    }
-
-    this.characterCounter_.setCounterValue(currentLength, maxLength);
-  }
-
   /**
    * Sets the aria label of the leading icon.
-   * @param {string} label
    */
-  setLeadingIconAriaLabel(label) {
+  setLeadingIconAriaLabel(label: string): void {
     if (this.leadingIcon_) {
       this.leadingIcon_.setAriaLabel(label);
     }
   }
 
   /**
    * Sets the text content of the leading icon.
-   * @param {string} content
    */
-  setLeadingIconContent(content) {
+  setLeadingIconContent(content: string): void {
     if (this.leadingIcon_) {
       this.leadingIcon_.setContent(content);
     }
   }
 
   /**
    * Sets the aria label of the trailing icon.
-   * @param {string} label
    */
-  setTrailingIconAriaLabel(label) {
+  setTrailingIconAriaLabel(label: string): void {
     if (this.trailingIcon_) {
       this.trailingIcon_.setAriaLabel(label);
     }
   }
 
   /**
    * Sets the text content of the trailing icon.
-   * @param {string} content
    */
-  setTrailingIconContent(content) {
+  setTrailingIconContent(content: string): void {
     if (this.trailingIcon_) {
       this.trailingIcon_.setContent(content);
     }
   }
 
   /**
-   * @return {boolean} True if the Text Field input fails in converting the
-   *     user-supplied value.
-   * @private
+   * Sets character counter values that shows characters used and the total character limit.
    */
-  isBadInput_() {
+  private setCharacterCounter_(currentLength: number): void {
+    if (!this.characterCounter_) return;
+
+    const maxLength = this.getNativeInput_().maxLength;
+    if (maxLength === -1) {
+      throw new Error('MDCTextFieldFoundation: Expected maxlength html property on text input or textarea.');
+    }
+
+    this.characterCounter_.setCounterValue(currentLength, maxLength);
+  }
+
+  /**
+   * user-supplied value
```

Comment:
Lost the first line of this parameter description too

Reply:
Done